src/qspdlog_model.cpp

```diff
@@ -22,20 +22,32 @@ static constexpr std::array<const char*, 3> column_names = { "Level",
 
 } // namespace
 
-QSpdLogModel::QSpdLogModel(QObject* parent)
+QSpdLogModel::QSpdLogModel(QObject* parent, int maxEntries)
     : QAbstractListModel(parent)
+    , _maxEntries(maxEntries)
 {
 }
 
 void QSpdLogModel::addEntry(entry_t entry)
 {
+    if(_maxEntries > 0 && _items.size() > _maxEntries) {
+        beginRemoveRows(QModelIndex(), 0, 0);
+        _items.erase(_items.begin());
```

Comment:
`erase`-ing from the beginning of the `std::vector` can lead to performance degradation.

Also, as you see I map the `index`'s `row` to the n-th item in the vector in the `data` function. So, removing from the beginning for the vector will lead to incorrect indexation.

Reply:
> Also, as you see I map the index's row to the n-th item in the vector in the data function. So, removing from the beginning for the vector will lead to incorrect indexation.

I'm not exactly sure what you mean by that. You index `_items` in the `data` method via `index.row()`. If I understand the [QAbstractListModel](https://doc.qt.io/qt-6/qabstractitemmodel.html#beginRemoveRows) documentation correctly, `beginRemoveRows()` should handle relocating the indices.

Sorry if I'm wrong, I fairly new to Qt.